File: BestTimeToBuyAndSellStock/maxProfit.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    var profit = 0;
    for(var i = 0; i < prices.length; i++) {
        var cur = 0;
        for(var j = i; j < prices.length; j++) {
            cur = Math.max(cur, prices[j] - prices[i]);
        }
        profit = Math.max(profit, cur);
    }
    return profit;
};

File: BestTimeToBuyAndSellStock/maxProfit.java
public class Solution {
    public int maxProfit(int[] prices) {
        int cur = 0; int res = 0;
        for(int i = 1; i < prices.length; i++) {
            cur = Math.max(0 , cur += prices[i] - prices[i - 1]);
            res = Math.max(cur, res);
        }
        return res;
    }
}